From ed4a6bd0c45cb08a57e7f4dbd2161569ded1b267 Mon Sep 17 00:00:00 2001
From: "brian.wagner" <brian.wagner@lairdtech.com>
Date: Mon, 7 Mar 2016 14:35:54 -0500
Subject: [PATCH] Add Guard Interval flag to data rate parsing

Section of Code which parsed Guard Interval was not correctly setting the GI flag.

bug 9139
---
 drivers/net/wireless/ath/ath6kl/wmi.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/ath/ath6kl/wmi.c b/drivers/net/wireless/ath/ath6kl/wmi.c
index c5cd5a4..f723a1a 100644
--- a/drivers/net/wireless/ath/ath6kl/wmi.c
+++ b/drivers/net/wireless/ath/ath6kl/wmi.c
@@ -3320,6 +3320,7 @@ struct ath6kl_rate_info ath6kl_wmi_get_rate(s8 rate_index)
 	/* SGI is stored as the MSB of the rate_index */
 	if (rate_index & RATE_INDEX_MSB) {
 		rate_index &= RATE_INDEX_WITHOUT_SGI_MASK;
+		info.sgi = 1;
 	}
 	/* Legacy Data Rate */
 	if (rate_index < ATH6KL_HT20_RATE_OFFSET) {
-- 
1.9.1

